source/BadgeOverlaySupport.js

```diff
@@ -0,0 +1,220 @@
+		* @private
+		*/
+		name: 'moon.BadgeOverlaySupport',
+
+		/**
+		* @private
+		*/
+		classes: 'moon-badge-overlay-support',
+
+		/**
+		* Identifies the MIME type of badge.
+		* Possible values are "image", "audio" or "video".
+		* Defaulted as 'image'.
+		*
+		* @name moon.BadgeOverlaySupport#badgeType
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+		badgeType: null,
+
+		/**
+		* Font icon to be used in place of default icon.
+		*
+		* @name moon.BadgeOverlaySupport#badgeIcon
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+		badgeIcon: null,
+
+		/**
+		* Url for icon to be used in place of default icon.
+		*
+		* @name moon.BadgeOverlaySupport#badgeSrc
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+		badgeSrc: null,
+
+		/**
+		* Sepcifies the alignment position of badge.
+		* Possible values are 'top' or 'bottom'.
+		* Defaulted as 'bottom'.
+		*
+		* @name moon.BadgeOverlaySupport#position
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+		position: null,
+
+		/**
+		* @private
+		*/
+		events: {
+
+			/**
+			* {@link moon.BadgeOverlaySupport#onBadgeTap}
+			*/
+			onBadgeTap: '',
+
+			/**
+			* {@link moon.BadgeOverlaySupport#onBadgeIconTap}
+			*/
+			onBadgeIconTap: ''
+		},
+
+		/**
+		* @method
+		* @private
+		*/
+		create: enyo.inherit(function (sup) {
+			return function () {
+				this.kindComponents = enyo.Component.overrideComponents(this.kindComponents, this._badgeOverride);
+				sup.apply(this, arguments);
+				this.positionChanged();
+				this.badgeTypeChanged();
+				if (this.badgeSrc) {
+					this.$.badgeScrimIcon.set('icon','');
+				}
+			};
+		}),
+
+		/**
+		* @private
+		*/
+		bindings: [
+			{from: 'badgeSrc', to: '$.badgeScrimIcon.src'},
+			{from: 'badgeIcon', to: '$.badgeScrimIcon.icon', transform: function(val){ return val ? val : 'play';}}
+		],
+
+		/**
+		* @private
+		*/
+		_badgeOverride: {
+			image: { kind:"moon.Image", components:[{
+					name:'badgeScrimIcon', kind: 'moon.Icon', small: true, icon: 'play', spotlight: false
+				}]
+			}
+		},
+
+		/**
+		* @private
+		*/
+		positionChanged: function() {
+			this.addRemoveClass("top", this.position == 'top');
+		},
+
+		/**
+		* @private
+		*/
+		badgeTypeChanged: function() {
+			var image = this.$.image;
+			image.destroyClientControls();
+			if(this.badgeType && this.badgeType != "image") {
+				image.createComponent(this._badgeOverride.image.components[0]);
+			}
+		},
+
+		/**
+		* @method
+		* @private
+		*/
+		dispatchEvent: enyo.inherit(function (sup) {
+			return function (sEventName, oEvent, oSender) {
+				if (oEvent && !oEvent.delegate) {
+					if (sEventName=='ontap'){
+						if(this.badgeTap(oSender, oEvent)) {
+							return true;
+						}
+					}
+				}
+				return sup.apply(this, arguments);
+			};
+		}),
+
+		/**
+		* @method
+		* @private
+		*/
+		badgeTap: function (inSender, inEvent) {
+			if (inEvent && inEvent.originator) {
+				switch(inEvent.originator.name) {
```

Comment:
I think we should revisit the event handling as this seems like a brittle approach. We're assuming a particular DOM structure that the mixin doesn't control. I couldn't get the onBadgeIconTap to fire at all because the tap area is the originator, not the icon. Further, we're breaking encapsulation by assuming the internal structure of moon.Image in referencing the `client` control.